clients/src/main/java/org/apache/kafka/common/requests/OffsetsForLeaderEpochRequest.java

```diff
@@ -158,4 +181,21 @@ public AbstractResponse getErrorResponse(int throttleTimeMs, Throwable e) {
         }
         return new OffsetsForLeaderEpochResponse(errorResponse);
     }
+
+    public static class PartitionData {
+        private final int currentLeaderEpoch;
+        public final int searchLeaderEpoch;
```

Comment:
I was not too happy with the name `searchLeaderEpoch`, so I'm happy to revert. I was trying to avoid confusion, but perhaps it is already clear enough since the current leader epoch is distinguished.